cpp/benchmarks/io/csv/csv_reader_options.cpp

```diff
@@ -60,43 +62,48 @@ void BM_csv_read_varying_options(
       .comment('#')
       .prefix("BM_");
 
-  size_t const chunk_size             = source_sink.size() / num_chunks;
-  cudf::size_type const chunk_row_cnt = view.num_rows() / num_chunks;
-  auto const mem_stats_logger         = cudf::memory_stats_logger();
+  size_t const chunk_size = cudf::util::div_rounding_up_safe(source_sink.size(), num_chunks);
+  auto const chunk_row_cnt =
+    cudf::util::div_rounding_up_safe(view.num_rows(), static_cast<cudf::size_type>(num_chunks));
```

Comment:
The old approach was rounding down and losing some rows. Adding the check uncovered the issue.
Also some logic in the loop got simplified by rounding up here.